src/utilities.js

```diff
@@ -157,3 +157,59 @@ export function createMelFilterBank(numFilters, sampleRate, bufferSize) {
 
   return filterBank;
 }
+
+export function hzToOctaves(freq, A440) {
+  return Math.log2(16 * freq / A440);
+}
+
+export function normalizeByColumn (a) {
+  var emptyRow = a[0].map(() => 0);
+  var colDenominators = a.reduce((acc, row) => {
+    row.forEach((cell, j) => {
+      acc[j] += Math.pow(cell, 2);
+    });
+    return acc;
+  }, emptyRow).map(Math.sqrt);
+  return a.map((row, i) => row.map((v, j) => v / (colDenominators[j] || 1) ));
+};
+
+export function createChromaFilterBank(numFilters, sampleRate, bufferSize, centerOctave=5, octaveWidth=2, baseC=true, A440=440) {
+  var numOutputBins = Math.floor(bufferSize / 2) + 1;
+
+
+  var frequencyBins = new Array(bufferSize).fill(0)
+    .map((_, i) => numFilters * hzToOctaves(sampleRate * i / bufferSize, A440));
+
+  // Set a value for the 0 Hz bin that is 1.5 octaves below bin 1
+  // (so chroma is 50% rotated from bin 1, and bin width is broad)
+  frequencyBins[0] = frequencyBins[1] - 1.5 * numFilters;
+
+  var binWidthBins = frequencyBins
+    .slice(1)
+    .map((v, i) => Math.max(v - frequencyBins[i]), 1)
+    .concat([1]);
+
+  var numFilters2 = Math.round(numFilters / 2);
```

Comment:
this variable name isn't very descriptive, why are there two `numFilters`? The name should denote the functional difference between the two